libs/template/reader.go

```diff
@@ -0,0 +1,155 @@
+package template
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/databricks/cli/libs/cmdio"
+)
+
+type Reader interface {
+	// FS returns a file system that contains the template
+	// definition files. This function is NOT thread safe.
+	FS(ctx context.Context) (fs.FS, error)
+
+	// Close releases any resources associated with the reader
+	// like cleaning up temporary directories.
+	Close() error
+}
+
+type builtinReader struct {
+	name     string
+	fsCached fs.FS
```

Comment:
Why cached? It's lazily initialized, so it's a regular property.

Stdlib uses `fsys` as variable name to avoid aliasing the package.

Reply:
The caching was important for the git reader, not really for this one. Since we only call `FS` once I have removed the caching from all readers.